# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFont.cpp
# Commit: e3f8526e0e2a
# Full Hash: e3f8526e0e2a08a4a8342dccedf05bebb3de1baf
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-09-08 21:39:05
# Regressor Bug: 1715501
# File Overlap Count: 1
# Description:
#   Bug 1715501 - patch 2 - Pass the presContext to platform font lookup methods so they can query it for font visibility. r=emilio
#   
#   This does not in itself change user-visible behavior, but is a foundation for bug
#   1715507 where we will make the behavior per-context instead of global. This is basically
#   just plumbing, passing a pointer to the presContext that's asking for fonts to be
# ==============================================================================

diff -r cfdd1c5d8d3f -r e3f8526e0e2a gfx/thebes/gfxFont.cpp
--- a/gfx/thebes/gfxFont.cpp	Wed Sep 08 09:35:16 2021 +0000
+++ b/gfx/thebes/gfxFont.cpp	Wed Sep 08 09:35:17 2021 +0000
@@ -3280,13 +3280,11 @@
     nsAtom* aLanguage, ShapedTextFlags aOrientation);
 
 template <>
-bool gfxFont::InitFakeSmallCapsRun(DrawTarget* aDrawTarget,
-                                   gfxTextRun* aTextRun, const char16_t* aText,
-                                   uint32_t aOffset, uint32_t aLength,
-                                   FontMatchType aMatchType,
-                                   gfx::ShapedTextFlags aOrientation,
-                                   Script aScript, nsAtom* aLanguage,
-                                   bool aSyntheticLower, bool aSyntheticUpper) {
+bool gfxFont::InitFakeSmallCapsRun(
+    nsPresContext* aPresContext, DrawTarget* aDrawTarget, gfxTextRun* aTextRun,
+    const char16_t* aText, uint32_t aOffset, uint32_t aLength,
+    FontMatchType aMatchType, gfx::ShapedTextFlags aOrientation, Script aScript,
+    nsAtom* aLanguage, bool aSyntheticLower, bool aSyntheticUpper) {
   bool ok = true;
 
   RefPtr<gfxFont> smallCapsFont = GetSmallCapsFont();
@@ -3435,19 +3433,18 @@
 }
 
 template <>
-bool gfxFont::InitFakeSmallCapsRun(DrawTarget* aDrawTarget,
-                                   gfxTextRun* aTextRun, const uint8_t* aText,
-                                   uint32_t aOffset, uint32_t aLength,
-                                   FontMatchType aMatchType,
-                                   gfx::ShapedTextFlags aOrientation,
-                                   Script aScript, nsAtom* aLanguage,
-                                   bool aSyntheticLower, bool aSyntheticUpper) {
+bool gfxFont::InitFakeSmallCapsRun(
+    nsPresContext* aPresContext, DrawTarget* aDrawTarget, gfxTextRun* aTextRun,
+    const uint8_t* aText, uint32_t aOffset, uint32_t aLength,
+    FontMatchType aMatchType, gfx::ShapedTextFlags aOrientation, Script aScript,
+    nsAtom* aLanguage, bool aSyntheticLower, bool aSyntheticUpper) {
   NS_ConvertASCIItoUTF16 unicodeString(reinterpret_cast<const char*>(aText),
                                        aLength);
-  return InitFakeSmallCapsRun(
-      aDrawTarget, aTextRun, static_cast<const char16_t*>(unicodeString.get()),
-      aOffset, aLength, aMatchType, aOrientation, aScript, aLanguage,
-      aSyntheticLower, aSyntheticUpper);
+  return InitFakeSmallCapsRun(aPresContext, aDrawTarget, aTextRun,
+                              static_cast<const char16_t*>(unicodeString.get()),
+                              aOffset, aLength, aMatchType, aOrientation,
+                              aScript, aLanguage, aSyntheticLower,
+                              aSyntheticUpper);
 }
 
 gfxFont* gfxFont::GetSmallCapsFont() {